### DISCUSSION TITLE
How to fix Cleartext HTTP traffic to 10.0.2.2 not permitted

### DISCUSSION THREAD

**Discussion by elibroftw** [unoplatform/uno#11484] [Q&A] [answered]
It would be good if the login or web API tutorial included how to consume a localhost API. 

I tried to add a network_security_config to the Resources/xml folder and here's my Android manifest but it's not respected. 

10.0.2.2 is the reference to localhost for the Android Emulator.

<details><summary>Android/Resources/xml/network_security_config.xml</summary>

```xml
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">10.0.2.2:44355</domain>
    </domain-config>
</network-security-config>
```

</details>

<details><summary>AndroidManifest.xml</summary>

```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android">
  <application android:networkSecurityConfig="@xml/network_security_config"></application>
</manifest>
```

</details>

**Comment by jeromelaban** [comment]
You may be able to find [more information here](https://stackoverflow.com/questions/67071052/how-to-fix-cleartext-http-traffic-to-x-not-permitted-in-xamarin-android).

**Comment by elibroftw** [comment, marked as answer]
Main.Androids.cs

```cs
namespace SplitTheTankApp.Droid {
    [Application(
        Label = "@string/ApplicationName",
        Icon = "@mipmap/icon",
        LargeHeap = true,
        HardwareAccelerated = true,
        Theme = "@style/AppTheme"
#if DEBUG
        , UsesCleartextTraffic = true
#endif
    )]
    public class Application : Microsoft.UI.Xaml.NativeApplication {
```
